packages/teleterm/src/ui/DocumentGateway/useDocumentGateway.ts

```diff
@@ -75,6 +73,12 @@ export default function useGateway(doc: types.DocumentGateway) {
     }
   }, [disconnectAttempt.status]);
 
+  useEffect(() => {
+    if (cluster.connected) {
```

Comment:
is optional chaining required here? just saw it on line 49

Reply:
Now that I look at it, it seems that the checks for cluster being null are unnecessary here. 🤔 If someone logs out from the cluster (and thus removes it) we remove any open resources from that cluster as well. We'll look into removing that after the preview release, thanks!